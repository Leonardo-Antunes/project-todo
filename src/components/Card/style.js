import styled from "styled-components";
import { Card, Chip, Typography } from '@material-ui/core'
import { CheckCircle } from "@material-ui/icons";

export const CardTodo = styled(Card)`
  width: 20vw;
  background-color: #fff;
  padding: 0.5rem 0.8rem;
  margin: 0.8rem;
`

export const CardTitle = styled(Typography)`
`;

export const CardDescription = styled(Typography)`
  line-height: 1.5rem;
  margin: 1rem 0;
  text-align: justify;
`;

export const TagBox = styled.div `
  display: flex;
  align-items: center;
  justify-content: space-between;
`

export const Tag = styled(Chip)`
  color: #fff;
  background-color: ${props => props.bgcolor};
`

export const Concluido = styled(CheckCircle)`
  color: #90be6d;
  font-size: 1.8rem;
`

export const NaoConcluido = styled(CheckCircle)`
  color: #f25c54;
  font-size: 1.8rem;
`