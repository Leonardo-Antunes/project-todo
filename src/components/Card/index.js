import { React, useState } from "react";
import { CardTitle, Wrapper, CardDescription, CardTag } from "./style";
import { useSelector, useDispatch } from "react-redux";
import inputActions from "../../redux/actions/inputActions";

const Card = ({ title, description, onItemClicked }) => {
  const [isChecked, setIsChecked] = useState(false);
  const dispatch = useDispatch();

  const handleIsChecked = (e) => {
    setIsChecked(!isChecked);
    dispatch(inputActions.setIsChecked(isChecked));
  };

  return (
    <Wrapper>
      <CardTitle>{title}</CardTitle>
      <CardDescription> {description} </CardDescription>
      {/* <CardTag> {tag} </CardTag> */}
      <button onClick={onItemClicked}>Texto</button>
      <input
        className="toggle"
        type="checkbox"
        checked={isChecked}
        onChange={handleIsChecked}
      />
      <CardTitle>{isChecked ? "oi" : "tchau"}</CardTitle>
    </Wrapper>
  );
};

export default Card;
