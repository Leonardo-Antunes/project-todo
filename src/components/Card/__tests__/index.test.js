import React from "react";
import Card from "../index";
import { render } from "@testing-library/react";

test("render div with text", () => {
  const { getByText } = render(<Card />);
  const textElement = getByText(/first test/i);
  expect(textElement).toBeInTheDocument();
});
