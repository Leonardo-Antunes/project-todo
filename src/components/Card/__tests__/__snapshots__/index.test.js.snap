// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root sc-bdnxRM eWUlBp MuiPaper-elevation1 MuiPaper-rounded"
>
  <h6
    className="MuiTypography-root sc-gtsrHT eOAXXw MuiTypography-h6"
  />
  <p
    className="MuiTypography-root sc-dlnjwi ixHxQZ MuiTypography-body2"
  >
     
     
  </p>
  <div
    className="sc-hKFxyN ccvTJW"
  >
    <div
      className="MuiChip-root sc-eCApnc jGYaxs"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
    >
      <span
        className="MuiChip-label"
      />
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root sc-gKAaRy juJupC"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
  </div>
</div>
`;
